app/src/main/java/com/mapbox/maps/testapp/examples/MapViewCustomizationActivity.kt

```diff
@@ -0,0 +1,84 @@
+package com.mapbox.maps.testapp.examples
+
+import android.os.Bundle
+import android.widget.LinearLayout
+import androidx.appcompat.app.AppCompatActivity
+import com.mapbox.geojson.Point
+import com.mapbox.maps.*
+import com.mapbox.maps.testapp.R
+import kotlinx.android.synthetic.main.activity_map_view_customization.*
+
+/**
+ * Example of customizing your [MapView].
+ * This example will show how to create [MapView] both programmatically and from xml
+ * and apply various options.
+ */
+class MapViewCustomizationActivity : AppCompatActivity() {
+
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+    setContentView(R.layout.activity_map_view_customization)
+    configureMapViewFromXml()
+    configureMapViewFromCode()
+  }
+
+  private fun configureMapViewFromCode() {
+    val mapboxMapOptions = MapboxMapOptions(this)
+    // set initial camera position
+    mapboxMapOptions.cameraOptions = CameraOptions.Builder()
+      .center(Point.fromLngLat(-122.4194, 37.7749))
+      .zoom(9.0)
+      .build()
+    // use texture view renderer
+    mapboxMapOptions.textureView = true
+    // set other map options
+    mapboxMapOptions.mapOptions = MapOptions.Builder()
+      .constrainMode(ConstrainMode.HEIGHT_ONLY)
+      .glyphsRasterizationOptions(
+        GlyphsRasterizationOptions.Builder()
+          .rasterizationMode(GlyphsRasterizationMode.IDEOGRAPHS_RASTERIZED_LOCALLY)
+          .build()
+      )
+      .build()
+    // set token and cache size for this particular map view
+    mapboxMapOptions.resourceOptions = ResourceOptions.Builder()
+      .accessToken(getString(R.string.mapbox_access_token))
+      .cacheSize(75_000L)
+      .build()
+    val customMapView = MapView(this, mapboxMapOptions)
+    val params = LinearLayout.LayoutParams(
+      LinearLayout.LayoutParams.MATCH_PARENT,
+      0,
+      1.0f
+    )
+    root.addView(customMapView, params)
+    customMapView.getMapboxMap().loadStyleUri(Style.SATELLITE)
+  }
+
+  private fun configureMapViewFromXml() {
+    // let's set `custom` token to MapView from code (however it will be same token from resources so that map will work)
+    MapboxOptions.setDefaultResourceOptions(this, getString(R.string.mapbox_access_token))
+    // all options provided in xml file - so we just load style
+    mapView.getMapboxMap().loadStyleUri(Style.DARK)
+  }
+
+  override fun onStart() {
+    super.onStart()
+    mapView.onStart()
```

Comment:
customMapView should also be handled in the life cycle methods

Reply:
you're totally correct!
thanks!